libmscore/segment.cpp

```diff
@@ -549,7 +549,11 @@ void Segment::add(Element* el)
                   _elist[track] = el;
                   _empty = false;
                   break;
-
+            case Element::Type::ANNOTATION:
+                  checkElement(el, track);
+                  _elist[track] = el;
```

Comment:
Done the necessary changes (including the layout call); however, putting a
breakpoint inside the ANNOTATION case for Segment::add, it is not being
called, which implies perhaps Segment::add is not being called for
annotation..

On Mon, Jun 20, 2016 at 9:26 AM, Marc Sabatella <notifications@github.com>
wrote:

> In libmscore/segment.cpp
> <https://github.com/musescore/MuseScore/pull/2666#discussion_r67633927>:
>
> > @@ -549,7 +549,11 @@ void Segment::add(Element* el)
> >                    _elist[track] = el;
> >                    _empty = false;
> >                    break;
> > -
> > +            case Element::Type::ANNOTATION:
> > +                  checkElement(el, track);
> > +                  _elist[track] = el;
>
> This isn't the right way to add an annotation. Look further up - see how
> staff texts (and pretty much all the other texts) are added, by adding to
> the "_annotations" list for the segment? You should probably just add this
> case to that list of other cases. Then the annotation will be in the list
> of segment annotations (I know, two different uses of the word annotation,
> confusing, but probably not worth changing at this point).
>
> Anyhow, once this is done, then the change I suggested to explicitly call
> layout when doing other segment annotations should work.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/musescore/MuseScore/pull/2666/files/42674354e00ad5ae833e4068eecc346c392352d4#r67633927>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AGf7ythHiQMpItYb7mUkckAvfyXuVDhzks5qNg-EgaJpZM4I1g-h>
> .
>



-- 
Ruchit Agrawal.
IIIT Hyderabad.